packages/shopify-api/lib/auth/oauth/types.ts

```diff
@@ -34,40 +34,42 @@ export interface OnlineAccessInfo {
   /**
    * The user associated with the access token.
    */
-  associated_user: {
-    /**
-     * The user's ID.
-     */
-    id: number;
-    /**
-     * The user's first name.
-     */
-    first_name: string;
-    /**
-     * The user's last name.
-     */
-    last_name: string;
-    /**
-     * The user's email address.
-     */
-    email: string;
-    /**
-     * Whether the user has verified their email address.
-     */
-    email_verified: boolean;
-    /**
-     * Whether the user is the account owner.
-     */
-    account_owner: boolean;
-    /**
-     * The user's locale.
-     */
-    locale: string;
-    /**
-     * Whether the user is a collaborator.
-     */
-    collaborator: boolean;
-  };
+  associated_user: OnlineAccessUser;
+}
+
+export interface OnlineAccessUser {
+  /**
+   * The user's ID.
+   */
+  id: number;
+  /**
+   * The user's first name.
+   */
+  first_name?: string;
```

Comment:
We currently have been using this OAuth type everywhere. 

From OAuth it will always have the full user information. 
When retrieving from storage, if they have not updated their session adapter then the additional user information will not be there.

To fix type errors with the tests I had to make these fields optional. Or should I make a new type for this?

Reply:
I added the new type in the latest commit. See if you are good with it!